src/simple-keyring.ts

```diff
@@ -0,0 +1,226 @@
+import { TxData, TypedTransaction } from '@ethereumjs/tx';
+import {
+  arrToBufArr,
+  bufferToHex,
+  ecsign,
+  isValidPrivate,
+  privateToPublic,
+  publicToAddress,
+  stripHexPrefix,
+  toBuffer,
+} from '@ethereumjs/util';
+import {
+  concatSig,
+  decrypt,
+  getEncryptionPublicKey,
+  normalize,
+  personalSign,
+  signTypedData,
+  SignTypedDataVersion,
+} from '@metamask/eth-sig-util';
+import { add0x, Eip1024EncryptedData, Hex, Keyring } from '@metamask/utils';
+import { keccak256 } from 'ethereum-cryptography/keccak';
+import randombytes from 'randombytes';
+
+const _TYPE = 'Simple Key Pair';
+
+export default class SimpleKeyring implements Keyring<string[]> {
+  #wallets: { privateKey: Buffer; publicKey: Buffer }[];
+
+  readonly type: string = _TYPE;
+
+  static type: string = _TYPE;
+
+  constructor(options: string[] = []) {
+    this.#wallets = [];
+
+    // istanbul ignore next
+    this.deserialize(options).catch(() => {
+      throw new Error('Problem deserializing SimpleKeyring');
```

Comment:
This hides the error along with its stack trace, which could be frustrating for debugging. Can we at least include the original error message in this string? Alternatively, we could consider using the `pony-cause` package along with a utility function to wrap the error so that we don't throw away the stack trace, as in here: https://github.com/MetaMask/create-release-branch/blob/main/src/misc-utils.ts#L83

Reply:
As I wrote in the other conversation, a constructor isn't allowed to be async, so this whole thing gets weird.  We should talk about options.

Reply:
Okay. If our plan is to make this better in the future, maybe we just include the original error in the message for now, something like this?

``` typescript
    this.deserialize(opts).catch((error) => {
      throw new Error('Problem deserializing SimpleKeyring', error);
    });
```